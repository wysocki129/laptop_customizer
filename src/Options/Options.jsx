import React, { Component } from 'react';
import './Options.css';

class Options extends Component {
	render() {
		const {
			allOptions,
			handleUpdateFeature,
			featureType,
			selectedOptions
		} = this.props;

		const featureOptions = allOptions.map((item, index) => {
			const selectedClass =
				item.name === selectedOptions[featureType].name
					? 'feature__selected'
					: ' ';
			const featureClass = 'feature__option ' + selectedClass;
			return (
				<li key={index} className="feature__item">
					<div
						onClick={e =>
							handleUpdateFeature(featureType, item)
						}
						className={featureClass}
					>
						{item.name}(
						{new Intl.NumberFormat('en-US', {
							style: 'currency',
							currency: 'USD'
						}).format(item.cost)}
						)
					</div>
				</li>
			);
		});

		return <> {featureOptions} </>;
	}
}
Options.defaultProps = {
	allOptions: []
};

export default Options;
