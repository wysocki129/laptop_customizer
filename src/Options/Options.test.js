import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Options from './Options';

const TESTOPTIONS = {
	Processor: [
		{
			name:
				'17th Generation Intel Core HB (7 Core with donut spare)',
			cost: 700
		},
		{
			name:
				'Professor X AMD Fire Breather with sidewinder technology',
			cost: 1200
		}
	]
};

describe('Options Component', () => {
	it('renders without crassing the app', () => {
		shallow(<Options />);
	});
	it('displays options when given', () => {
		const wrapper = shallow(<Options props={TESTOPTIONS} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
