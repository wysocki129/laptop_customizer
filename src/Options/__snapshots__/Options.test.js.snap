// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Component displays options when given 1`] = `
<Fragment>
   
   
</Fragment>
`;
